scripts/check-binary.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+BASE_URL="https://binaries.soliditylang.org/bin"
+REPO_ROOT="$(dirname "$0")/.."
+LIST_FILE=/tmp/list.json
+
+function fail() {
+    echo -e "ERROR: $@" >&2
+    exit 1
+}
+
+function check_release_version() {
+    current_version=$1
+
+    curl --silent --fail "$BASE_URL/list.json" -o $LIST_FILE
+    [[ ! -f $LIST_FILE ]] && fail "Download of release list failed:\n    [url]: $BASE_URL/list.json"
+
+    # Retrieve the latest released version
+    latest_version=$(cat $LIST_FILE | jq --raw-output ".latestRelease")
+    release_version=$(cat $LIST_FILE | jq --raw-output ".releases | .[\"$latest_version\"]" | sed --regexp-extended --quiet 's/^soljson-v(.*).js$/\1/p')
```

Comment:
I think it would be more correct to get the version from the `longVersion` field of the `builds` list instead of parsing it out of the path. Technically, the path could be arbitrary and we just happen to put it in the file name but we don't have to.

Reply:
The problem is that the `latestRelease` is a short version, and I don't know beforehand what is the correspondent long version of the release to be able to disambiguate in the builds list. So I needed to extract that information from the `releases` field and filter the correct entry in the `builds`.

Reply:
But yes, I guess it is better to invert the logic, try to find the current release (of which I know the long version) in the builds list, as you said, and then see if its short version is the latest release.

Reply:
Good point about it being ambiguous. I did not notice that.

But in that case you can still use the path you get from `.releases` to find the right entry in `.builds` and avoid parsing that path with a regex. Paths are supposed to be unique and it will be more robust.

Reply:
The point there was to compare only the long versions between `current_version` and `latest_version_long` ignoring some path elements like `soljson-v` or `.js`. But I applied the changes you suggested and used the `path` to avoid the extra regex.